linkcheck/checker/httpurl.py

```diff
@@ -78,15 +78,11 @@ def content_allows_robots (self):
         """
         if not self.is_html():
             return True
-        # construct handler object
-        handler = linkparse.MetaRobotsFinder()
-        # parse
-        try:
-            htmlsoup.process_soup(handler, self.get_soup())
-        except linkparse.StopParse as msg:
-            log.debug(LOG_CHECK, "Stopped parsing: %s", msg)
-            pass
-        return handler.follow
+
+        if self.get_soup().find("meta", attrs={"name": "robots",
+                                               "content": "nofollow"}):
+            return False
+        return True
```

Comment:
Our test suite has a single file with

```
<meta name="robots" content="nofollow">
```

so I think we need to add more tests that have variations (`content="noindex, nofollow, nosnippet"` etc), including bad ones (`<meta name="robots">` without `content=` etc).

Ideally I wouldn't want each of those tests to require an entire test file -- talk about overhead!  How about simple unit tests that overwrite some attributes of a local url_data variable store this or that bit of HTML instead of having it go and fetch it from the network?

Reply:
```py
        soup = self.get_soup()
        return not soup.find(..
```
That's better, it did look ugly.

```py
return 'nofollow' not in meta_robots.get('content', '')
```
.find() takes a regexp so I have gone with:
```py
re.compile(r"\bnofollow\b", re.IGNORECASE)
```

> a local url_data variable store this or that bit of HTML instead of having it go and fetch it from the network?

I think the complication is that it is a method of a HttpUrl object (grandchild of UrlBase) which takes a bit of initialising. I've made the example in the file more complex for now,  could add another commit later.

Reply:
Can copy the object creation at least from e.g. test_internpat.py.